Snowflake.Data.Tests/IntegrationTests/VectorTypesIT.cs

```diff
@@ -0,0 +1,346 @@
+        {
+            using (DbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+                AlterSessionSettings(conn);
+
+                using (DbCommand command = conn.CreateCommand())
+                {
+                    command.CommandText = $"SELECT [{float.MinValue}, {float.MaxValue}]::VECTOR(FLOAT, 2) as vec;";
+                    var reader = (SnowflakeDbDataReader)command.ExecuteReader();
+
+                    Assert.IsTrue(reader.Read());
+
+                    var arr = reader.GetArray<float>(0);
+#if NETFRAMEWORK
+                    Assert.AreEqual(float.MinValue, arr[0], 4.1E+31d);
+                    Assert.AreEqual(float.MaxValue, arr[1], 4.1E+31d);
+#else
+                    Assert.AreEqual(float.MinValue, arr[0]);
+                    Assert.AreEqual(float.MaxValue, arr[1]);
+#endif
+                }
+                conn.Close();
+            }
+        }
+
+        [Test]
+        public void TestSelectFloatVectorWithNoDecimals()
+        {
+            using (DbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+                AlterSessionSettings(conn);
+
+                using (DbCommand command = conn.CreateCommand())
+                {
+                    command.CommandText = "SELECT [1,2,3]::VECTOR(FLOAT, 3) as vec;";
+                    var reader = (SnowflakeDbDataReader)command.ExecuteReader();
+
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual("[1.000000,2.000000,3.000000]", reader.GetString(0));
+
+                    var arr = reader.GetArray<float>(0);
+                    Assert.AreEqual(1, arr[0]);
+                    Assert.AreEqual(2, arr[1]);
+                    Assert.AreEqual(3, arr[2]);
+                }
+                conn.Close();
+            }
+        }
+
+        [Test]
+        public void TestSelectFloatVectorWithGreaterThanSixDigitPrecision()
+        {
+            using (DbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+                AlterSessionSettings(conn);
+
+                using (DbCommand command = conn.CreateCommand())
+                {
+                    command.CommandText = "SELECT [1.123456789,2.123456789,3.123456789]::VECTOR(FLOAT, 3) as vec;";
+                    var reader = (SnowflakeDbDataReader)command.ExecuteReader();
+
+                    Assert.IsTrue(reader.Read());
+                    Assert.AreEqual("[1.123457,2.123457,3.123457]", reader.GetString(0));
+
+                    var arr = reader.GetArray<float>(0);
+                    Assert.AreEqual(1.123456789, arr[0], 0.1);
+                    Assert.AreEqual(2.123456789, arr[1], 0.1);
```

Comment:
Even after adding the `f` to the value, the test still fails with a difference of 0.00000011
```
Expected: 1.12345684f
But was:  1.12345695f
```

In the docs, they include an example of [comparing float values](https://learn.microsoft.com/en-us/dotnet/api/system.single.equals?view=net-8.0#system-single-equals(system-object)) and including tolerance under the Remarks section

Reply:
the server responded with float numbers rounded to 6 digits precision. You even made an assert:
```
Assert.AreEqual("[1.123457,2.123457,3.123457]", reader.GetString(0));
```
asserting that the result was `[1.123457,2.123457,3.123457]`

So I think you should just assert that:
```
Assert.AreEqual(1.123457f, arr[0]);
Assert.AreEqual(2.123457f, arr[1]);
Assert.AreEqual(3.123457f, arr[2]);
```

Reply:
Ah I see, I was doing an assert with the original value `Assert.AreEqual(1.123456789f, arr[0]);` instead of the rounded values

Changed the expected values to the rounded values